web/src/store/repos.ts

```diff
@@ -1,44 +1,54 @@
 import { defineStore } from 'pinia';
-import { computed, Ref, toRef } from 'vue';
+import { computed, reactive, Ref, ref } from 'vue';
 
 import useApiClient from '~/compositions/useApiClient';
 import { Repo } from '~/lib/api/types';
 import { repoSlug } from '~/utils/helpers';
 
-const apiClient = useApiClient();
-
-export default defineStore({
-  id: 'repos',
-
-  state: () => ({
-    repos: {} as Record<string, Repo>,
-  }),
-
-  actions: {
-    // getter
-    getRepo(owner: Ref<string>, name: Ref<string>) {
-      return computed(() => {
-        const slug = repoSlug(owner.value, name.value);
-        return toRef(this.repos, slug).value;
-      });
-    },
-
-    // setter
-    setRepo(repo: Repo) {
-      this.repos[repoSlug(repo)] = repo;
-    },
-
-    // loading
-    async loadRepo(owner: string, name: string) {
-      const repo = await apiClient.getRepo(owner, name);
-      this.repos[repoSlug(repo)] = repo;
-      return repo;
-    },
-    async loadRepos() {
-      const repos = await apiClient.getRepoList();
-      repos.forEach((repo) => {
-        this.repos[repoSlug(repo.owner, repo.name)] = repo;
-      });
-    },
-  },
+export const useRepoStore = defineStore('repos', () => {
+  const apiClient = useApiClient();
+
+  const repos: Map<string, Repo> = reactive(new Map());
+  const ownedRepoSlugs = ref<string[]>([]);
+
+  const ownedRepos = computed(() =>
+    Array.from(repos.entries())
+      .filter(([slug]) => ownedRepoSlugs.value.includes(slug))
+      .map(([, repo]) => repo),
+  );
+
+  function getRepo(owner: Ref<string>, name: Ref<string>) {
+    return computed(() => {
+      const slug = repoSlug(owner.value, name.value);
+      return repos.get(slug);
+    });
+  }
+
+  function setRepo(repo: Repo) {
+    repos.set(repoSlug(repo), repo);
+  }
+
+  async function loadRepo(owner: string, name: string) {
+    const repo = await apiClient.getRepo(owner, name);
+    repos.set(repoSlug(repo), repo);
+    return repo;
+  }
+
+  async function loadRepos() {
+    const _repos = await apiClient.getRepoList();
+    _repos.forEach((repo) => {
+      repos.set(repoSlug(repo), repo);
+    });
+    ownedRepoSlugs.value = _repos.map((repo) => repoSlug(repo));
```

Comment:
```suggestion
    const _ownedRepos = await apiClient.getRepoList();
    _ownedRepos.forEach((repo) => {
      repos.set(repoSlug(repo), repo);
    });
    ownedRepoSlugs.value = _ownedRepos.map((repo) => repoSlug(repo));
```

@anbraten What do you think of this?